### DISCUSSION TITLE
Iniciando em programação front-end, o que indicam? Quais os melhores cursos?

### DISCUSSION THREAD

**Discussion by julia-droid** [frontendbr/forum#1578] [Discussão]
Pessoal, estou começando a estudar programação, vendo bastante sobre html, css e javascript, mas quero também já ir na direção de aprender react, que vejo estar bem forte...

Encontro muita coisa na internet e nas documentações, mas queria algo que tivesse uma linha de raciocínio e aprendizado para eu seguir, vou continuar estudando pela documentação e testes também, mas gostaria da indicação de cursos que ajudaram de verdade vocês, seja em html, css e js, como também em react e desenvolvimento geral web...

Quais cursos ou materiais indicam?

Obrigada

**Comment by wilsonneto-swe** [comment]
Se tratando de HTML, CSS e JS acredito que o W3School é muito bom e gratuito, foi por ali e na prática que aprendi.

Já com ReactJS eu fiz alguns mas indico apenas dois, que me ajudaram de verdade e valeram muito o investimento:

OneBitCode - FullStack JS
Um ótimo curso para quem está começando e quer focar o inicio de carreira em javascript, aprenda desde o básico até questões avançadas e arquiteturais sobre Node, ReactJS e features avançadas do JS/ES. O melhor caminho para quem está começando e quer se tornar um fullstack. Um ótimo custo benefício.

https://bit.ly/onebitcode-fullstack-js](https://bit.ly/onebitcode-fullstack-js

DevPleno - ReactJS
Um curso com um ótimo custo benefício, focado em ReactJS, ideal para aqueles que querem iniciar sua carreira focados em front-end, em algumas situações é interessante focar os estudos no frontend, e quando dominar o frontend seguir para aprender backend. É um ótimo curso que aborda do básico ao avançado, arquitetura, testes, tudo que é necessário para dominar o front.

https://bit.ly/reactjs-dev-pleno](https://bit.ly/reactjs-dev-pleno


Mas claro, a documentação da tecnologia é a base, mas cursos são bem bons para praticar com projetos prontos como a semana agora da rocketseat.

Apenas indicando, pois em ajudaram muito a chegar onde estou como fullstack, entregando projetos em produção pela empresa sem medo e com muita segurança...

Abraços

**Reply by danielrl76** [reply]
"Mas claro, a documentação própria é a base,..." O que seria a "documentação própria"??? Também sou principiante, então apreciaria muito a melhor explicação que você puder dar sobre isso e o que mais você achar importante e que tenha relação. Grato!

**Reply by vitordefante** [reply]
Acho que ele se refere a documentação de cada tecnologia/linguagem. Por exemplo, no site do **Python** tem a documentação deles que basicamente é a explicação de uso da linguagem toda, uma espécie de **manual**.

**Reply by danielrl76** [reply]
Lendo de novo concordo com você Vítor. Na minha primeira leitura eu havia entendido "documentação própria" como sendo documentação feita pelo dev de seus projetos, isto é, "própria do dev". Mas ele quis dizer isso mesmo que você disse, a documentação da linguagem, framework, etc, sendo estudada ou empregada (o que eu já sabia ser a prática correta desde que me interessei por desenvolvimento), "própria DA LINGUAGEM". Vlw!

**Comment by gustavocarmoh** [comment]
Eu sou desenvolvedor React, e tenho uma dica para lhe dar.

Acredito que não disponha logo de cara de uma verba para poder estar investindo tanto em cursos e equipamentos, eu lhe recomendo antes de fazer um investimento em um curso como por exemplo **RocketSeat** (diga-se de passagem tenho muita vontade de fazer) , **Dev Pleno** ou **Alura**, procure alternativas gratuitas ou mais baratas.

Por exemplo, existem vários tutoriais/cursos no youtube disponibilizados de forma gratuita:

- Curso de JavaScript -> https://www.youtube.com/playlist?list=PLHz_AreHm4dlsK3Nr9GVvXCbpQyHQl1o1
- Curso de HTML 5 -> https://www.youtube.com/playlist?list=PLHz_AreHm4dlAnJ_jJtV29RFxnPHDuk9o

Caso disponha de uma quantia pequena de dinheiro, recomendo os cursos da Udemy, são cursos bons e mais acessíveis.

Agora mais importante do que um curso, é aprender o Inglês, se você não sabe ainda invista nisso, depois que você possua uma noção básica de programação se torna muito fácil aprender as coisas, porem é necessário saber o Inglês para poder ler documentações e alguns tutoriais, infelizmente a maioria dos conteúdos é feito em inglês.

Espero ter conseguido ajudar um pouco.
Lhe desejo bons estudos!

**Reply by Emri-commin** [reply]
É o curso  que estou fazendo agora .
Ele tem um didática plausível, não tem como você não entender. Ele mastiga o assunto todinho. O melhor na minha opinião.

**Comment by ViniciusGularte** [comment]
@julia-droid  Bem vinda ao mundo front-end! :rocket:

Alguns recursos que podem te ajudar nesta trilha que me ajudaram: 

[FreeCodeCamp](https://www.freecodecamp.org/learn) - Te oferece um mapa completo de front-end pra vc seguir, onde você finaliza os desafios e ganha até certificado pelo mesmo ( tudo gratuito)

[Javascript desafio 30 dias](https://javascript30.com/) - 30 dias de desafios com Javascript

Na minha opinião, conceitos, padrões de projetos, arquiteturas valem mais do que o hype do momento, eu trabalho com React atualmente, mas pense que antes de querer aprender React aprenda o problema que ele vai solucionar pra vc, pense que a um tempo atrás jQuery tava no hype, hoje é React, qual sera o próximo ? Como vc mesmo disse que está começando a estudar programação não sugiro que estude estes  frameworks logo de cara, e sim a linguagem por trás deles(Javascript :heart:  ). 

Alguns conteúdos a mais pra vc olhar :


[Eloquente Javascript](https://github.com/braziljs/eloquente-javascript)

[Padrões de projetos com Javascript](https://github.com/kamranahmedse/design-patterns-for-humans)

[30 Segundos de código](https://github.com/30-seconds/30-seconds-of-code)

[Desafios Front End](https://github.com/felipefialho/frontend-challenges) - Pro futuro ( depois que tiver pegado as manhas )

Tem muito mais coisa, daria pra criar um repositório só de links, mas isto você vai vendo no dia a dia, sucesso na área :rocket:

**Comment by wendeleyf** [comment]
Olá @julia-droid !
Caso você tenha condições de conseguir o [Github Student Pack](https://education.github.com/pack), além de todas as dicas que todos deram aqui, você pode acessar durante 6 meses os cursos do [Frontend Masters](https://frontendmasters.com) onde é possível encontrar cursos sobre todo ecosistema Javascript.

Pode ser uma boa fonte de estudos também!

**Comment by ihenrique** [comment]
Da uma lida nisso: [https://github.com/kamranahmedse/developer-roadmap](https://github.com/kamranahmedse/developer-roadmap).
Só toma cudiado, fornt-end é um mundo maluco e sem foco cada dia tem uma framework nova...não se esqueça dos 3 pilares ( HTML, CSS, Javascript ), não adianta vc ter uma ferrari sem carta de motorista!

**Comment by felipefialho** [comment]
Olá @julia-droid, tudo bem?

Esses dias escrevi um artigo sobre [O que Front-end Developers precisam saber](https://www.felipefialho.com/blog/o-que-front-end-developers-precisam-saber/), está bem completinho.

Como está começando sua jornada, vale dar uma olhada.

**Comment by sscraul** [comment]
Pessoal, muito obrigado pelas dicas!
Alguém sabe se os cursos do TreinaWeb são bons? O site parece muito bem organizado, alguém já fez e recomenda?
No momento estou aprendendo lógica da programação, depois vou iniciar JS e HTML/CSS.
Grande abraço.

**Comment by felipefialho** [comment]
Saindo do forno 🍕 

Veja se te ajuda https://www.felipefialho.com/blog/como-estudar-praticar-e-conseguir-oportunidades-como-front-end-developer/

**Comment by jpferreiradev** [comment]
Felipe, com certeza você é a minha grande inspiração no mundo front, no back é o Fábio Akita! kkkkkk! Muito obrigado por cada artigo o//

**Comment by julia-droid** [comment]
Oi pessoal, passando mais para agradecer a todos, @felipefialho seu artigo e seu blog em geral ajudou demais!!

Obrigado a todos pelas opiniões, e também ao @wilsonneto-dev  pela indicação do curso, estou fazendo atualmente o curso do Dev Pleno ([http://bit.ly/reactjs-dev-pleno](http://bit.ly/reactjs-dev-pleno)) e indico muito, achei a didádica bem boa e estou conseguindo sentir a evolução... 

E consegui uma oportunidade de estágio seguindo as dicas e estudando muito \o/

**Comment by RafaelSSantiago** [comment]
@julia-droid  iai tudo bem? me conta com quanto tempo de estudo vc conseguiu seu estágio? como foi a entrevista? parabéns pela conquista

**Comment by Fabio-Pina** [comment]
> Se tratando de HTML, CSS e JS acredito que o W3School é muito bom e gratuito, foi por ali e na prática que aprendi.
> 
> Já com ReactJS eu fiz alguns mas indico apenas dois, que me ajudaram de verdade e valeram muito o investimento:
> 
> Bootcamp da RocketSeat (sai por R$ 1980,00, um pouco elevado mas deu retorno, porém fecharam as inscrições e vão abrir na próxima semana omnistack)
> https://rocketseat.com.br/gostack-9/inscricao
> 
> E se achou muito caro tem também o curso do dev pleno que fiz antes até do da rocketseat e ele me ajudou a conquistar o cargo de fullstack que tenho hoje, e é bem mais em conta.
> 
> Curso Dev Pleno que ta saindo R$ 697,00:
> http://bit.ly/reactjs-dev-pleno
> 
> Mas claro, a documentação própria é a base, mas cursos são bem bons para praticar com projetos prontos como a semana agora da rocketseat.
> 
> Apenas indicando, pois em ajudaram muito a chegar onde estou como fullstack, entregando projetos em produção pela empresa sem medo e com muita segurança...
> 
> Abraços

Olá Wilson, eu estou estudando pela W3C Schools e estou achando bem legal. Eu acho que tem uma coisa ou outra desatualizada mas a base tem lá inclusive em várias linguagens. Estou estudando FrontEnd também. Tem vários exercícios legais por lá. Eu vou ler aqui sobre o Rockseat! Ah, eu lhe segui aqui no GitHub, obrigado e abraço.

**Comment by wilsonneto-swe** [comment]
Booa, a thread cresceu rs

Só para deixar mais uma indicação, o pessoal da one bit code fez um novo curso focado em desafios para aprendizado e para entrevistas, achei bem massa... compartilhando: https://bit.ly/desafios-para-entrevistas
